src/Symfony/Component/Ldap/Security/User/LdapUser.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Ldap\Security\User;
+
+use Symfony\Component\Ldap\Entry;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ *
+ * @internal
+ */
+class LdapUser implements UserInterface
+{
+    private $entry;
+    private $username;
+    private $password;
+    private $roles;
+    private $extraFields;
+
+    public function __construct(Entry $entry, string $username, ?string $password, array $roles = [], array $extraFields = [])
+    {
+        if (!$username) {
+            throw new \InvalidArgumentException('The username cannot be empty.');
+        }
+
+        $this->entry = $entry;
+        $this->username = $username;
+        $this->password = $password;
+        $this->roles = $roles;
+        $this->extraFields = $extraFields;
+    }
+
+    public function getEntry(): Entry
+    {
+        return $this->entry;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRoles()
+    {
+        return $this->roles;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPassword()
+    {
+        return $this->password;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSalt()
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getUsername()
+    {
+        return $this->username;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function eraseCredentials()
+    {
+    }
```

Comment:
Shouldn't this be clearing the password? Wouldn't want it to end up in a session